lib/ansible/modules/system/systemd.py

```diff
@@ -307,6 +307,114 @@ def parse_systemctl_show(lines):
     return parsed
 
 
+def unit_is_enabled(module, systemctl_executable, systemd_unit, is_initd):
```

Comment:
make module and systemctl globals to avoid repassing them all the time,

is_initd should not be passed in as it would be for the original service and not the ones you depend on (not sure its possible to mix those in systemd)

Reply:
I avoid globals, it's just bad. If I wanted shared state I'd go for a different approach which requires refactoring and is out of the scope of this PR.

Reply:
Regarding initd, AFAIK init.d scripts usually just point to systemd units... I can make the adjustment, though.

Reply:
Globals are 'bad' in persistent environments, in this case 'one of scripts' that are designed to end on execution, they become a lot more efficient than passing objects all over.

As for init.d pointing to units are more of an exception than the norm, debian for example ships with initial init.d on services to allow people to switch out of systemd. Many other services don't have systemd units but do have the traditional init scripts.